### DISCUSSION TITLE
Providing Packages

### DISCUSSION THREAD

**Discussion by mytlogos** [TechnitiumSoftware/DnsServer#720] [Ideas]
The current install ways for linux are:

- Use the docker image
- Download and run a bash script
- Install the Portable Version and manually install the ASP.Net Runtime (either via package or as tar)
- Build it yourself

I personally dislike installing critical services like dns, dhcp etc. as docker images, as docker is more likely to break/restart etc. in comparison to a manually/systemd managed service.
But it is far easier and self-contained.

Running a bash script downloaded from a network is relatively popular? but it always looks like a security smell to me.

I could install manually by writing my own script or just (always) review your install script before executing it, but it does seem like a hassle.

So i am asking if you maybe want to provide deb/rpm packages manually or via an repository, which can facilitate the un-/installation by integration with the linux package ecosystem.

There are tools for creating packages automatically

- [fpm](https://github.com/jordansissel/fpm) - no "official" dotnet support? though i do not know if it is required
- [dotnet-packaging](https://github.com/quamotion/dotnet-packaging) - may not have dotnet 7 support?
- [dotnet-make-deb](https://github.com/ygoe/DotnetMakeDeb) - does not seem active?
- ...

and there are ways to host package repositories
- ppa for debs
- [Aptly](https://www.aptly.info/) for self hosting deb (only?) packages
- ...

The dependency on the asp.net runtime could be "solved" by providing a package with the runtime bundled and not adding it to the path (where other dotnet runtimes may lurk around).

In the same sense, one could provide packages for windows with chocolatey.

Providing packages probably does add some serious overhead, depending on how one does it, though, so i am only asking if you ever have the intention or would "outsource" it to whomever.

Opinions?

**Comment by ShreyasZare** [comment]
Thanks for asking.

> Providing packages probably does add some serious overhead, depending on how one does it, though, so i am only asking if you ever have the intention or would "outsource" it to whomever.

As you mentioned, it is an overhead to maintain packages especially since there are many Linux distros which use different package managers. The team for this entire project is just me developing the software as and when I can find some time after by day job. So, may be in future this can be done but it would be too difficult to manage it at the moment.

**Reply by mytlogos** [reply]
Thanks for the answer.
